app/javascript/controllers/funnel_chart_controller.js

```diff
@@ -51,11 +55,31 @@ export default class extends Controller {
             }
           }
         },
-        //spanGaps: true,
         plugins: {
           legend: {
              display: this.displayLegend(),
              position: 'bottom'
+          },
+          datalabels: {
+            labels: {
+              title: {
+                color: 'white',
+                font: {
+                  weight: 'bold',
+                  lineHeight: 1.3
+                },
+                textAlign: 'center'
+              }
+            },
+            formatter: (value, context) => {
+              let testValue = this.funnelDataValue.map((x, y, z) => Math.round(x/z[y-1]*100));
+              testValue[0] = '';
+              if (context.dataIndex == 0) {
+                return value
+              } else {
+                return value + '\n' + '(' + testValue[context.dataIndex] + '%)'
+              }
```

Comment:
The `else` is not necessary, since you're returning in the previous if branch anyways. You can just put the return statement on the next line after the if.

It's better to use string interpolation than concatenating strings with `+`.

Reply:
yeah but with string interpolation the \n gave a weird big space in the text in the bar chart, so I used this method instead